Added in basic Docker support and cleaned up the Readme

robwagner33 opened the pull request:
Nothing fancy here. Just a basic docker setup and some instructions.

Whiteaj36 committed:
Initial commit

robwagner33 committed:
Added docker support and updated the Readme

robwagner33 committed:
Added in a new logo

petitJAM commented on Dockerfile:
```diff
@@ -0,0 +1,8 @@
+FROM ruby:2.3.3
```
the `.ruby-version` file is using 2.2.5.  idk what this does but it seems like it should match.  we were going to use 2.4.0, but there's some bug with 2.4.0 on macs.  you could probably just update the ruby-version

robwagner33 replied:
Thats the environment its going to be based on (just a 2.3.3 ruby development env). We can change it to whatever. I'll update to 2.2.5

robwagner33 replied:
If you want, we can target ruby 2.4.0. With docker, the mac issue would be irrelevant. Currently our web container is based on a debian image.

robwagner33 replied:
Let's plan on targeting 2.2.5 unless we find a compelling reason for 2.4

robwagner33 committed:
Updated to use ruby 2.2.5

petitJAM reviewed (approved):
:shipit: I'll try out this docker stuff later

Whiteaj36 commented:
What does Docker do?

Whiteaj36 commented:
Read through their intro. Still curious about the difference between what was added and removed before and after this pr

robwagner33 commented:
I added the Dockerfile, which is what docker uses to build the web image.

I added the docker_compose yaml, which is what we use to do the composition of our app. If you look at the yaml its telling the DB container to pull from a pre-existing postgresql image, and telling the web container to build '.' which is whats defined in that Dockerfile.

Then I updated the readme with new instructions and changed the database config for the app to use our new 'db' instance.

Instead of running 'rails s' anymore, you can use the new instructions to work with docker. This will help us to map whats in production more closely with your development environment. Eventually I'll publish the docker image so that you dont need to build it yourself.

Whiteaj36 commented:
So if I have an existing migration waiting I'll need to rake to get this new db up to date?

robwagner33 commented:
Yeah anytime you have a migration you need to run, you have to run rake db:migrate to get the DB and schema up to date.

Feel free to ping me or alex about activerecord migrations and stuff. Honestly probably need a better place to chat about this rather than on an old PR
